src/index.ts

```diff
@@ -53,21 +59,38 @@ class Webhooks<TTransformed = unknown> {
       log: createLogger(options.log),
     };
 
+    if ("path" in options) {
+      state.log.warn(
+        "[@octokit/webhooks] `path` option is deprecated and will be removed in a future release of `@octokit/webhooks`. Please use `createNodeMiddleware(webhooks, { path })` instead"
+      );
+    }
+
     this.sign = sign.bind(null, options.secret);
     this.verify = verify.bind(null, options.secret);
     this.on = state.eventHandler.on;
     this.onAny = state.eventHandler.onAny;
     this.onError = state.eventHandler.onError;
     this.removeListener = state.eventHandler.removeListener;
     this.receive = state.eventHandler.receive;
-    this.middleware = middleware.bind(null, state);
     this.verifyAndReceive = verifyAndReceive.bind(null, state);
+
+    this.middleware = function deprecatedMiddleware(
+      request: IncomingMessage,
+      response: ServerResponse,
+      next?: (err?: any) => void
+    ) {
+      state.log.warn(
+        "[@octokit/webhooks] `webhooks.middleware` is deprecated and will be removed in a future release of `@octokit/webhooks`. Please use `createNodeMiddleware(webhooks)` instead"
+      );
+      return middleware(state, request, response, next);
+    };
   }
 }
 
 /** @deprecated `createWebhooksApi()` is deprecated and will be removed in a future release of `@octokit/webhooks`, please use the `Webhooks` class instead */
 const createWebhooksApi = <TTransformed>(options: Options<TTransformed>) => {
-  console.error(
+  const log = createLogger(options.log);
+  log.warn(
```

Comment:
This isn't strictly related to this PR, but I get the changes

Reply:
yeah sorry for that, you are totally right, I've been lazy creating separate PRs for that. Same with the formatting changes in the README, I've to show `octokit` on Wednesday, I'm in a bit of a rush